File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        for (int num: nums) {
            freq[num]++;
        }
        vector<vector<int>> buckets(nums.size() + 1);
        for (auto& entry: freq) {
            int num = entry.first;
            int count = entry.second;
            buckets[count].push_back(num);
        }

        vector<int> result;
        for(int i = buckets.size() - 1; i>=0; i--) {
            for (int num: buckets[i]) {
                result.push_back(num);
                if(result.size() == k) {
                    return result;
                }
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        q=1
        for i in range(1, len(nums)):
            val = nums[i-1]
            if nums[i] != val:
                nums[q] = nums[i]
                q +=1

        return q